cesm/nuopc_cap_share/init_pio_mod.F90

```diff
@@ -583,90 +395,12 @@ subroutine shr_pio_getiotypefromname(typename, iotype, defaulttype)
     else if ( typename .eq. 'DEFAULT') then
        iotype = defaulttype
     else
-       write(shr_log_unit,*) 'shr_pio_mod: WARNING Bad io_type argument - using iotype_netcdf'
+       write(shr_log_unit,*) 'init_pio_mod: WARNING Bad io_type argument - using iotype_netcdf'
        iotype=pio_iotype_netcdf
     end if
 
-  end subroutine shr_pio_getiotypefromname
-
-!===============================================================================
-  subroutine shr_pio_namelist_set(npes,mycomm, pio_stride, pio_root, pio_numiotasks, &
```

Comment:
I have removed shr_pio_namelist_set: this appears to be unused and was buggy (in particular, it referenced the old module-level scalar `pio_async_interface`, but that variable was never set; now I have removed that scalar and replaced it with an array, as I describe in a separate comment).